### PR TITLE
Fix py.typed in namespace packages

### PR DESCRIPTION
**Pull request by srittau**
Previously, the py.typed file was incorrectly put in the namespace package, instead of the non-namespace sub-packages.

### CODE REVIEW

**Comment by srittau** [comment]
Cc @Avasam

**Comment by srittau** [comment]
Could someone review this? The google packages are still broken.

File: tests/test_integration.py [hunk 1]
```diff
@@ -169,3 +164,35 @@ def test_verify_requires_python() -> None:
...
+        ("Pillow", ["PIL-stubs"]),
+        ("protobuf", ["google-stubs.protobuf"]),
+        ("google-cloud-ndb", ["google-stubs.cloud.ndb"]),
+    ],
+)
+def test_pkg_data_non_namespace_packages(
+    distribution: str, expected_packages: list[str]
+) -> None:
+    pkg_data = build_wheel.collect_package_data(THIRD_PARTY_PATH / distribution)
+    assert pkg_data.top_level_non_namespace_packages == expected_packages
```
Comment (Avasam) [inline tests/test_integration.py:198]: I just noticed, I think there's no test to validate that `py.typed` is added to the right location(s). Could we add unit tests for the `if metadata.partial` branch of `generate_setup_file` calling `add_partial_markers`?

**Review by Avasam — CHANGES_REQUESTED** [review]
I have one optional suggestion and one request for an additional test concerning `py.typed` files. Otherwise LGTM.

File: tests/test_integration.py (hunk 1 above)
Comment (srittau) [reply tests/test_integration.py:198]: Good idea, but I think it makes more sense as part of #121, considering the current command/query mixture.

**Review by Avasam — APPROVED** [review]

**Comment by srittau** [comment]
Unfortunately, it seems that py.typed is now dropped completely from the package. I'll look into it later.

**Comment by srittau** [comment]
It's been a few weeks, but I also remember testing this locally where it seemed to work. Probably best to just work on #121 instead of trying to find this bug.

**Comment by Avasam** [comment]
@srittau Should I wait before merging PRs relating to partial stubs? (ie https://github.com/python/typeshed/pull/11255/files ) 
iirc would affect type-checking users that also use modules missing from typeshed.

**Comment by srittau** [comment]
I would actually be interested if it also affects partial non-namespace packages.

**Comment by Avasam** [comment]
> I would actually be interested if it also affects partial non-namespace packages.

pyopenssl was released 20h ago and contains the marker file
https://pypi.org/project/types-pyOpenSSL/#history
image

I believe this PR was merged 29h ago?

If that's still too close/unsure, you'll know for sure once tensorflow (modified 3h ago) deploys.

---

I think I understand what you're thinking of: maybe only namespace packages are missing the `py.typed` file. Which could be caused, for example, by an incorrect path added to `package_data[package]`

**Pull request merged**